subsys/dfu/dfu_target/src/dfu_target_suit.c

```diff
@@ -134,26 +170,66 @@ int dfu_target_suit_done(bool successful)
 
 int dfu_target_suit_schedule_update(int img_num)
 {
+	/* img_num is not needed for SUIT */
+	(void)img_num;
+
+	int err;
 	LOG_INF("Schedule update");
 
-	suit_plat_mreg_t update_candidate[] = {
-		{
-			.mem = DFU_PARTITION_ADDRESS,
-			.size = image_size,
-		},
-	};
+	/* Process the saved envelope */
+	err = suit_dfu_candidate_preprocess();
```

Comment:
There are few similar places that can be easily optimized by one line.
```suggestion
	int err = suit_dfu_candidate_preprocess();
```